connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -364,11 +365,31 @@ private static String castToString(Object value) {
         if (value instanceof java.util.Date) {
             java.util.Date dateValue = (java.util.Date) value;
             return Values.dateFormatFor(dateValue).format(dateValue);
+        } else if (value instanceof ByteBuffer) {
+            ByteBuffer byteBuffer = (ByteBuffer) value;
+            if (byteBuffer.hasArray()) {
+                return castByteArrayToString(byteBuffer.array());
+            }
+            else {
+                byte[] array = new byte[byteBuffer.remaining()];
+                byteBuffer.get(array);
+                return castByteArrayToString(array);
+            }
```

Comment:
The use of `ByteBuffer.get(...)` here does not account for the fact that it may not be positioned at the beginning. Kafka has two `Utils.readBytes(...)` methods that we should probably use. This code would then simplify to:
```suggestion
            byte[] rawBytes = Utils.readBytes(byteBuffer);
            return castByteArrayToString(rawBytes);
```

Reply:
That sounds like an excellent plan, I did not know this class existed. Changed.

Reply:
This sounds like an excellent plan and passes the tests.